stats.go

```diff
@@ -458,6 +459,27 @@ func newTypedValue(vd *view.View, r *view.Row) *monitoringpb.TypedValue {
 	return nil
 }
 
+func shouldInsertZeroBound(bounds ...float64) bool {
+	if len(bounds) > 0 && bounds[0] != 0.0 {
```

Comment:
Should we check if the first bound is negative?

Reply:
I think no need to check for negative or zero bucket bounds, as we drop these bounds at top level API. I am not sure how it's done in Go, you may want to refer Java's implementation -> https://github.com/census-instrumentation/opencensus-java/blob/17e327f21921f981951608727602203bbc82238a/api/src/main/java/io/opencensus/stats/BucketBoundaries.java#L64-L90 and https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java#L153-L156

Reply:
@songy23  negative value is checked [here](https://github.com/census-instrumentation/opencensus-go/blob/950a67f393d867cfbe91414063b69e511f42fefb/stats/view/view.go#L97-L101)

@mayurkale22 the reason for checking 0 value is that you could have different version (older version) of core library which would send 0 bound bucket. So you need to insert it conditionally.

Reply:
SGTM